backend/services/user_groups.go

```diff
@@ -0,0 +1,441 @@
+				Slug: slugMap[j].GroupSlug,
+				Name: slugMap[j].GroupName,
+				UserSlugs: []string{
+					slugMap[j].UserSlug.String,
+				},
+				Deleted: slugMap[j].Deleted.Valid,
+			}
+		} else if firstItem && noUserSlug {
+			tempGroupMap = dtos.UserGroupAdminView{
+				Slug:    slugMap[j].GroupSlug,
+				Name:    slugMap[j].GroupName,
+				Deleted: slugMap[j].Deleted.Valid,
+			}
+		} else if otherItem && sameGroupAsPrev && hasUserSlug {
+			tempGroupMap.UserSlugs = append(tempGroupMap.UserSlugs, slugMap[j].UserSlug.String)
+		} else if otherItem && diffGroup && hasUserSlug {
+			userGroupsDTO = append(userGroupsDTO, tempGroupMap)
+			tempGroupMap = dtos.UserGroupAdminView{
+				Slug: slugMap[j].GroupSlug,
+				Name: slugMap[j].GroupName,
+				UserSlugs: []string{
+					slugMap[j].UserSlug.String,
+				},
+				Deleted: slugMap[j].Deleted.Valid,
+			}
+		} else if otherItem && diffGroup && noUserSlug {
+			userGroupsDTO = append(userGroupsDTO, tempGroupMap)
+			tempGroupMap = dtos.UserGroupAdminView{
+				Slug:    slugMap[j].GroupSlug,
+				Name:    slugMap[j].GroupName,
+				Deleted: slugMap[j].Deleted.Valid,
+			}
+		} else if isLastItem && sameGroupAsPrev && hasUserSlug {
+			tempGroupMap.UserSlugs = append(tempGroupMap.UserSlugs, slugMap[j].UserSlug.String)
+			userGroupsDTO = append(userGroupsDTO, tempGroupMap)
+		} else if isLastItem && sameGroupAsPrev && noUserSlug {
+			userGroupsDTO = append(userGroupsDTO, tempGroupMap)
+		} else if isLastItem && diffGroup && hasUserSlug {
+			userGroupsDTO = append(userGroupsDTO, tempGroupMap)
+			tempGroupMap = dtos.UserGroupAdminView{
+				Slug: slugMap[j].GroupSlug,
+				Name: slugMap[j].GroupName,
+				UserSlugs: []string{
+					slugMap[j].UserSlug.String,
+				},
+				Deleted: slugMap[j].Deleted.Valid,
+			}
+			userGroupsDTO = append(userGroupsDTO, tempGroupMap)
+		} else if isLastItem && diffGroup && noUserSlug {
+			userGroupsDTO = append(userGroupsDTO, tempGroupMap)
+			tempGroupMap = dtos.UserGroupAdminView{
+				Slug:    slugMap[j].GroupSlug,
+				Name:    slugMap[j].GroupName,
+				Deleted: slugMap[j].Deleted.Valid,
+			}
+			userGroupsDTO = append(userGroupsDTO, tempGroupMap)
+		}
+	}
+
+	prevLastIndex := (pagination.Page - 1) * pagination.PageSize
+	groupLength := len(userGroupsDTO)
+	totalPages := math.Ceil(float64(groupLength) / float64(pagination.PageSize))
+	remainingItemsCount := (groupLength - int(prevLastIndex)) % int(pagination.PageSize)
+
+	currLastIndex := int(pagination.Page * pagination.PageSize)
+	pageSize := pagination.PageSize
+	if pagination.Page == int64(totalPages) {
+		currLastIndex = int(prevLastIndex) + remainingItemsCount
+		pageSize = int64(remainingItemsCount)
+	}
+
+	paginatedResults := userGroupsDTO[prevLastIndex:currLastIndex]
+	paginatedData := &dtos.PaginationWrapper{
+		PageNumber: pagination.Page,
+		PageSize:   pageSize,
+		Content:    paginatedResults,
+		TotalCount: int64(groupLength),
+		TotalPages: int64(totalPages),
+	}
+	return paginatedData, nil
```

Comment:
I can't rely on the normal pagination, because I'm receiving a table that lists each user and their affiliated group AND a row that just lists the table, in case there are no users assigned to the group, which returns way more rows than I need, so I had to write my own pagination logic

Reply:
uuuuuh, yeah. Maybe let's drop pagination? Or let's just stick with what you have here? (I have a slight preference for the former --  if we have to maintain this later, then I think we'd be better off with a simple, unoptimized solution).